WorkspaceServer.Tests/Kernel/CSharpKernelTests.cs

```diff
@@ -209,12 +221,32 @@ public async Task it_produces_a_final_value_if_the_code_expression_evaluates()
 5", "csharp");
             await kernel.SendAsync(kernelCommand);
 
-            KernelEvents.OfType<ValueProduced>()
+            KernelEvents
+                .Where(k => k.Value is ValueProduced)
+                .Select(k => k.Value as ValueProduced)
                 .Should()
                 .HaveCount(4)
                 .And
                 .ContainSingle(e => e.IsLastValue);
-              
+
+        }
+
+        [Fact]
+        public async Task the_output_is_asynchronous()
+        {
+            var kernel = CreateKernel();
+
+            var kernelCommand = new SubmitCode(@"
+Console.Write(DateTime.Now);
+System.Threading.Thread.Sleep(1000);
```

Comment:
Using a custom scheduler, we wouldn't need an actual `Thread.Sleep` and the test can run faster.